File: Downloads/testSimple/api/upload-file.js
export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(405).json({ error: "Method not allowed" });
  }

  try {
    const { fileName, fileContentBase64 } = req.body;

    const response = await fetch(
      "https://dbc-1b7ba151-05e9.cloud.databricks.com/api/2.0/dbfs/put",
      {
        method: "POST",
        headers: {
          Authorization: `Bearer ${process.env.DATABRICKS_TOKEN}`,
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          path: `/FileStore/uploads/${fileName}`,
          contents: fileContentBase64,
          overwrite: true,
        }),
      }
    );

    const data = await response.json();
    res.status(200).json({ success: true, databricks: data });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
}

File: Downloads/testSimple/api/upload_file.py
import json
import os
import requests

def handler(request):
    if request.method != "POST":
        return {
            "statusCode": 405,
            "body": json.dumps({"error": "Method not allowed"})
        }

    try:
        body = request.json()
        file_name = body.get("fileName")
        file_content_base64 = body.get("fileContentBase64")

        databricks_token = os.environ.get("DATABRICKS_TOKEN")

        url = "https://dbc-1b7ba151-05e9.cloud.databricks.com/api/2.0/dbfs/put"

        payload = {
            "path": f"/FileStore/uploads/{file_name}",
            "contents": file_content_base64,
            "overwrite": True
        }

        headers = {
            "Authorization": f"Bearer {databricks_token}",
            "Content-Type": "application/json"
        }

        response = requests.post(url, headers=headers, json=payload)

        return {
            "statusCode": 200,
            "body": json.dumps({
                "success": True,
                "databricks_response": response.json()
            })
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)})
        }
